pkg/services/authn/grpcutils/config.go

```diff
@@ -0,0 +1,45 @@
+package grpcutils
+
+import (
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type Mode string
+
+func (s Mode) IsValid() bool {
+	switch s {
+	case ModeOnPrem, ModeCloud:
+		return true
+	}
+	return false
+}
+
+const (
+	ModeOnPrem Mode = "on-prem"
+	ModeCloud  Mode = "cloud"
+)
+
+type GrpcServerConfig struct {
+	SigningKeysURL   string
+	AllowedAudiences []string
+	Mode             Mode
+	LegacyFallback   bool
+}
+
+func ReadGrpcServerConfig(cfg *setting.Cfg) (*GrpcServerConfig, error) {
+	section := cfg.SectionWithEnvOverrides("grpc_server_authentication")
+
```

Comment:
Rather than setting mode explicitly... Can this be based on stackId !=""?


IIUC, this is all temp until we use access tokens everywhere, so if we can avoid changes to settings that would be best

Reply:
So this is the server side, does the mt resource store have a stackID?

Reply:
The MT resource store doesn't have a stackID no.

Reply:
Would that work: We could derive the mode directly from the StorageType https://github.com/grafana/grafana/blob/main/pkg/services/apiserver/options/storage.go#L21
If StorageTypeUnifiedGrpc, then we are in Cloud.

Reply:
Also, isn't `StorageTypeUnifiedGrpc` used to test things locally as well?